xarray/test/test_quantities.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import pandas as pd
+
+from xarray import (align, broadcast, Dataset, DataArray, Variable)
+
+from xarray.test import (TestCase, unittest)
+
+try:
+    import quantities as pq
+    has_quantities = True
+except ImportError:
+    has_quantities = False
+
+def requires_quantities(test):
+    return test if has_quantities else unittest.skip('requires python-quantities')(test)
+
+@requires_quantities
+class TestWithQuantities(TestCase):
+    def setUp(self):
+        self.x = np.arange(10) * pq.A
+        self.y = np.arange(20)
+        self.xp = np.arange(10) * pq.J
+        self.v = np.arange(10*20).reshape(10,20) * pq.V
+        self.da = DataArray(self.v,dims=['x','y'],coords=dict(x=self.x,y=self.y,xp=(['x'],self.xp)))
+
+    def assertEqualWUnits(self,a,b):
+        self.assertIsNotNone(getattr(a, 'units', None))
+        self.assertIsNotNone(getattr(b, 'units', None))
+        self.assertEqual(a.units,b.units)
+        np.testing.assert_allclose(a.magnitude,b.magnitude)
+
+    def test_units_in_data_and_coords(self):
+        da = self.da
+        self.assertEqualWUnits(da.xp.data,self.xp)
+        self.assertEqualWUnits(da.data,self.v)
```

Comment:
Can we adjust things so that `DataArray.values` still always returns a base `numpy.ndarray`, but `DataArray.data` can return a subclass? That would be consistent with our current behavior for handling duck-type arrays with dask.